File: src/hasoftware-library/src/hasoftware/api/messages/NotifyRequest.java
package hasoftware.api.messages;

import hasoftware.api.FunctionCode;
import hasoftware.api.Message;
import hasoftware.cdef.CDEFMessage;
import java.util.ArrayList;
import java.util.List;

public class NotifyRequest extends Message {

    private final List<Integer> _functionCodes = new ArrayList<>();

    public NotifyRequest() {
        super(FunctionCode.Notify, 0, 0);
    }

    public NotifyRequest(CDEFMessage cdefMessage) {
        super(cdefMessage);
        int count = cdefMessage.getU8();
        for (int i = 0; i < count; i++) {
            int functionCode = cdefMessage.getU16();
            _functionCodes.add(functionCode);
        }
    }

    @Override
    public void encode(CDEFMessage cdefMessage) {
        super.encode(cdefMessage);
        cdefMessage.putU8(_functionCodes.size());
        for (int count = 0; count < _functionCodes.size(); count++) {
            cdefMessage.putU16(_functionCodes.get(count));
        }
    }

    public List<Integer> getFunctionCodes() {
        return _functionCodes;
    }
}

File: src.net/hasoftware-library/Messages/NotifyRequest.cs
using hasoftware.Api;
using hasoftware.Cdef;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace hasoftware.Messages
{
    public class NotifyRequest : Message
    {
        public List<int> FunctionCodes { get; private set; }

        public NotifyRequest()
            : base (Api.FunctionCode.Notify, 0, 0)
        {
            FunctionCodes = new List<int>();
        }

        public NotifyRequest(CdefMessage cdefMessage)
            : base(cdefMessage)
        {
            FunctionCodes = new List<int>();
            int count = cdefMessage.GetU8();
            for (int i = 0; i < count; i++) {
                FunctionCodes.Add(cdefMessage.GetU16());
            }
        }

        public override void Encode(CdefMessage cdefMessage) {
            base.Encode(cdefMessage);
            cdefMessage.PutU8(FunctionCodes.Count);
            foreach (var functionCode in FunctionCodes) {
                cdefMessage.PutU16(functionCode);
            }
        }
    }
}
